pkg/op/storage.go

```diff
@@ -50,6 +50,15 @@ type AuthStorage interface {
 	GetKeySet(context.Context) (*jose.JSONWebKeySet, error)
 }
 
+// CanRefreshTokenInfo is an optional additional interface that Storage can support.
+// Supporting CanRefreshTokenInfo is required to be able to revoke a refresh token that
+// does not happen to also be JWTs.
```

Comment:
this irritates me...
the zitadel implementation also uses refresh tokens that are not JWTs, but rather opaque strings (as probably most implementations do)
and in fact, if I don't implement the interface, the token will still be decrypted

Reply:
I'm open to other solutions. 

The problem is in this function:
```go
func getTokenIDAndSubjectForRevocation(ctx context.Context, userinfoProvider UserinfoProvider, accessToken string) (string, string, bool) {
        tokenIDSubject, err := userinfoProvider.Crypto().Decrypt(accessToken)
        if err == nil {
                splitToken := strings.Split(tokenIDSubject, ":")
                if len(splitToken) != 2 {
                        return "", "", false
                }
                return splitToken[0], splitToken[1], true
        }
        accessTokenClaims, err := VerifyAccessToken(ctx, accessToken, userinfoProvider.AccessTokenVerifier())
        if err != nil {
                return "", "", false
        }
        return accessTokenClaims.GetTokenID(), accessTokenClaims.GetSubject(), true
}
```

Prior to my change, it gets called with refresh tokens.  Randomly, some refresh tokens will successfully decrypt.   Randomly, some of the decrypted refresh tokens will have exactly one ":" in them.

Reply:
OK i get the intention. And the current way is naive (<- couldn't think of a better word)
It really makes no sense to just assume that the token is either an opaque token (either access or refresh and both an encrypted form of <tokenID>:<userID>) or a JWT
Maybe we just need to change the description to:
```
// CanRefreshTokenInfo is an optional additional interface that Storage can support.
// Supporting CanRefreshTokenInfo is required to be able to (revoke) a refresh token that
// is neither an encrypted string of <tokenID>:<userID> nor a JWT.
```
Or is this confusing?

Reply:
It's a little confusing, but it's clear enough.  I've applied that comment.